codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Decoder.java

```diff
@@ -110,11 +110,11 @@ private void reset() {
     /**
      * Decode the header block into header fields.
      */
-    public void decode(InputStream in, HeaderListener headerListener) throws IOException {
-        while (in.available() > 0) {
+    public void decode(ByteBuf in, Headers<CharSequence, CharSequence, ?> headers) throws IOException {
```

Comment:
I think we should require `Http2Headers` instead of `Headers<CharSequence, CharSequence, ?>`.